Improving Parsing SDF File

minhnq1297 opened the pull request:
- Using ElementTree to parse and resolve SDF file instead of doing it recursively

- Adding inertial elements if they are missing from SDF file

minhnq1297 committed:
Improving Parsing SDF File

- Using ElementTree to parse and resolve SDF file instead of doing it
recursively

- Adding inertial elements if they are missing from SDF file

ARoefer commented on src/iai_bullet_sim/utils.py:
```diff
@@ -166,29 +176,45 @@ def abs_urdf_paths(file_path, temp_dir):
+        if inertial is None:
+            inertial = ET.SubElement(link, 'inertial')
+            mass = ET.SubElement(inertial, 'mass')
+            mass.text = str(0)
```
Is assuming 0 mass a good idea? That makes everything static.

minhnq1297 replied:
I think it's not a good idea for robot description. I'll change it to a small number, though it might get conflict with possible existing inertia moment. But I agree that without inertia moment, it could be moved as a particle, but without mass, it stands still.

minhnq1297 replied:
And I think in this case, if `inertial` component is missing, we assume that the object is static. But I still changed it to small default value. Please take a look at my new commit

ARoefer commented:
The `load_sdf` function should also work for `package://` paths

minhnq1297 committed:
Improving Parsing SDF File

- Changing default value for mass and inertia when loading SDF

- Adapt default value of moment of inertia according to SDF document

- Supporting loading SDF from ROS 'package://'

ARoefer commented on src/iai_bullet_sim/utils.py:
```diff
@@ -166,29 +186,45 @@ def abs_urdf_paths(file_path, temp_dir):
+                inertia = ET.SubElement(inertial, 'inertia')
+                for k in ['ixx', 'ixy', 'ixz', 'iyy', 'iyz', 'izz']:
+                    i = ET.SubElement(inertia, k)
+                    i.text = str(0.01)
```
Please change the mass to 1. The sdf specification assumes that as default weight, if nothing else is specified: (http://sdformat.org/spec?ver=1.10&elem=link)

ARoefer commented:
Except for the default mass, I am good with this commit and would merge it.
